src/utils/viewer/viewer.cpp

```diff
@@ -769,6 +798,44 @@ void Viewer::drawEvent() {
   redraw();
 }
 
+void Viewer::moveAndLook(int repetitions) {
+  if (repetitions <= 2) {repetitions = 1;}    // for visual smoothness, for small deviations around 60fps, we default to repetitions=1.
+
+  for (int i = 0; i < repetitions; i++) {
+    if (keysPressed[KeyEvent::Key::Left]) {
+      defaultAgent_->act("turnLeft");
+    }
+    if (keysPressed[KeyEvent::Key::Right]) {
+      defaultAgent_->act("turnRight");
+    }
+    if (keysPressed[KeyEvent::Key::Up]) {
+      defaultAgent_->act("lookUp");
+    }
+    if (keysPressed[KeyEvent::Key::Down]) {
+      defaultAgent_->act("lookDown");
+    }
+
+    if (keysPressed[KeyEvent::Key::A]) {
+      defaultAgent_->act("moveLeft");
+    }
+    if (keysPressed[KeyEvent::Key::D]) {
+      defaultAgent_->act("moveRight");
+    }
+    if (keysPressed[KeyEvent::Key::S]) {
+      defaultAgent_->act("moveBackward");
+    }
+    if (keysPressed[KeyEvent::Key::W]) {
+      defaultAgent_->act("moveForward");
+    }
+    if (keysPressed[KeyEvent::Key::X]) {
+      defaultAgent_->act("moveDown");
+    }
+    if (keysPressed[KeyEvent::Key::Z]) {
+      defaultAgent_->act("moveUp");
+    }
```

Comment:
recAgentLocation() should be called in here, as in the original movement cases.  At the end of this section, if any of the movement actions have executed (if the agent's location has changed in any way), recAgentLocation() should be called.